src/api/java/com/minecolonies/api/research/effects/IResearchEffect.java

```diff
@@ -38,11 +38,31 @@
      */
     TranslationTextComponent getDesc();
 
+    /**
+     * Effect subtitle description.
+     *
+     * @return the Subtitle desc.
+     */
+    String getSubtitle();
+
+    /**
+     * Returns the effect's display value
+     *
+     * @return the display value.
+     */
+    Double getDisplay();
+
     /**
      * Does this effect override another effect with the same id?
      *
      * @param other the effect to check.
      * @return true if so.
      */
     boolean overrides(@NotNull final IResearchEffect<?> other);
```

Comment:
That's been one of the challenges for the data-driven effects PR-to-PR you asked for.

Where a numeric output exists these comparisons aren't terribly hard to work with, but for tagged functions that overrides method may well get removed.  I've left it for now because a) it existed and b) it will provide some protection against packmaker created research trees where an effect of the same type exists on multiple different branches or subbranches, but, yes, the data-driven effects will probably see it disappear.